### PR TITLE
Add client version check API call, closes MISC-300

### PR DESCRIPTION
**Pull request by guilgaly**
Did this one first as a simple example, but will need to be finished once the API is done since it's not publicly documented.

### CODE REVIEW

File: src/main/java/io/gatling/plugin/util/LambdaExceptionUtil.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.function.Function;
+
+final class LambdaExceptionUtil {
+
+    @FunctionalInterface
+    public interface FunctionWithExceptions<T, R, E extends Exception> {
+        R apply(T t) throws E;
+    }
+
+    public static <T, R, E extends Exception> Function<T, R> rethrowFunction(FunctionWithExceptions<T, R, E> function) throws E {
```
Comment (tpetillot) [inline src/main/java/io/gatling/plugin/util/LambdaExceptionUtil.java]: `rethrowFunction` does not throws E? function does but is transformed as unchecked.

File: src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java [hunk 2]
```diff
@@ -23,67 +23,109 @@
...
-        .addPathSegment("artifacts")
-        .addPathSegment(artifactId.toString())
-        .addPathSegment("content")
-        .addQueryParameter("filename", file.getName())
-        .build();
+  @Override
+  public void checkVersionSupport(String client, String version)
+      throws UnsupportedClientException, EnterpriseClientException {
+    Request.Builder request = checkVersionSupportRequest(client, version);
+    // TODO see how this API call should respond
```
Comment (tpetillot) [inline src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java]: Is there a reference issue?

File: src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java [hunk 3]
```diff
@@ -23,67 +23,109 @@
...
+        throw new UnsupportedClientException(client, version);
+    }
   }
 
-  private Request uploadPackageRequest(UUID artifactId, File file) {
-    HttpUrl url = uploadPackageUrl(artifactId, file);
-    RequestBody body = RequestBody.create(OCTET_STREAM_MEDIA_TYPE, file);
-    return new Request.Builder().url(url).put(body).header(AUTHORIZATION_HEADER, token).build();
+  private Request.Builder checkVersionSupportRequest(String client, String version) {
+    // TODO update this API call url/parameters
```
Comment (tpetillot) [inline src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java]: Same

File: src/main/java/io/gatling/plugin/util/LambdaExceptionUtil.java (hunk 1 above)
Comment (tpetillot) [reply src/main/java/io/gatling/plugin/util/LambdaExceptionUtil.java]: Not used?

**Review by tpetillot — CHANGES_REQUESTED** [review]

File: src/main/java/io/gatling/plugin/util/AbstractApiRequests.java [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+
+  void defaultValidateResponse(Response response) throws EnterpriseClientException {
+    if (!response.isSuccessful()) {
+      switch (response.code()) {
+        case HttpURLConnection.HTTP_UNAUTHORIZED:
+          throw new EnterpriseClientException(
+              "Your API token was not recognized by the Gatling Enterprise server: please configure a valid token");
+        case HttpURLConnection.HTTP_FORBIDDEN:
+          throw new EnterpriseClientException(
+              "Your API token does not have enough privileges: please configure a token with the correct role");
```
Comment (ccedric) [inline src/main/java/io/gatling/plugin/util/AbstractApiRequests.java:78]: We could mention here that the API token needs the "All" role

Comment (slandelle) [reply src/main/java/io/gatling/plugin/util/AbstractApiRequests.java:78]: That, or add a link to the precise point in the documentation. But we don't want to have messages that basically mean "now go dig in the documentation, glhf".

File: src/main/java/io/gatling/plugin/util/LambdaExceptionUtil.java (hunk 1 above)
Comment (guilgaly) [reply src/main/java/io/gatling/plugin/util/LambdaExceptionUtil.java]: No longer used after I re-refactored some stuff, exact. 👍

(This was a trick for throwing checked exceptions within a lamba that is passed to a method expecting a standard function type from `java.util.function.*`, e.g. to `Stream#map()`, as those function types of course don't declare checked exceptions.)

File: src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java (hunk 2 above)
Comment (guilgaly) [reply src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java]: Can be done once https://gatlingcorp.atlassian.net/browse/CLOUD-867 is done.

File: src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java (hunk 3 above)
Comment (guilgaly) [reply src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java]: Can be done once https://gatlingcorp.atlassian.net/browse/CLOUD-867 is done.

File: src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java (hunk 2 above)
Comment (tpetillot) [reply src/main/java/io/gatling/plugin/util/OkHttpEnterpriseClient.java]: Please refer the issue 👍

**Review by tpetillot — APPROVED** [review]

**Review by ccedric — APPROVED** [review]

**Pull request merged**